app/controllers/katello/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -295,6 +300,12 @@ def find_deletable_hosts
       find_bulk_hosts(:destroy_hosts, params)
     end
 
+    def validate_organization
```

Comment:
This controller includes a concern which defines a method called `find_organization`

https://github.com/Katello/katello/blob/2fd5de8ae422881bf457c1d5d359f6a87836e803/app/controllers/katello/concerns/api/v2/bulk_hosts_extensions.rb#L40-L42

Perhaps you can update that method to use some memoization:

```
      def find_organization
        @organization ||= Organization.find_by_id(params[:organization_id])
      end
```